### PR TITLE
must-gather: removes hardcoded subvolumegroup

### PR DESCRIPTION
**Pull request by nik-redhat**
**Description:**
Removed the hardcoded svg, and added command to fetch the list of
subvolumegroups for each filesystem and get the subvols, snapshots for
each of them

Fixes: #1596

Signed-off-by: nik-redhat <nladha@redhat.com>

### CODE REVIEW

File: must-gather/collection-scripts/gather_ceph_resources [hunk 1]
```diff
@@ -332,23 +332,27 @@ for ns in $namespaces; do
...
-                 for snap in $snaps; do
-                     { printf "Information for snapshot: %s\n" "${snap}" >> "${COMMAND_OUTPUT_FILE}"; }
-                     { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "ceph fs subvolume snapshot info $fs $subvol $snap $svg --connect-timeout=15" >> "${COMMAND_OUTPUT_FILE}"; } >> "${COMMAND_ERR_OUTPUT_DIR}"/gather-ceph-fs-"${subvol}"-"${snap}"-debug.log 2>&1 &
-                     pids_ceph+=($!)
-                 done
-             done
+            # Get the subvolumegroup names from the filesystem name
+            ceph_command="ceph fs subvolumegroup ls ${fs}"
+            subvolgrp_name=$(timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "${ceph_command}"|awk -F\" '{print $4}')
+            for $svg in $subvolgrp_name; do
```
Comment (Madhu-1) [inline must-gather/collection-scripts/gather_ceph_resources:339]: `for $svg in $subvolgrp_name; do` to `for svg in $subvolgroups; do`

File: must-gather/collection-scripts/gather_ceph_resources [hunk 2]
```diff
@@ -332,23 +332,27 @@ for ns in $namespaces; do
...
-                 { printf "Snapshot count in subvolume: %s=%s\n" "${subvol}" "${count}" >> "${COMMAND_OUTPUT_FILE}"; }
-                 for snap in $snaps; do
-                     { printf "Information for snapshot: %s\n" "${snap}" >> "${COMMAND_OUTPUT_FILE}"; }
-                     { timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "ceph fs subvolume snapshot info $fs $subvol $snap $svg --connect-timeout=15" >> "${COMMAND_OUTPUT_FILE}"; } >> "${COMMAND_ERR_OUTPUT_DIR}"/gather-ceph-fs-"${subvol}"-"${snap}"-debug.log 2>&1 &
-                     pids_ceph+=($!)
-                 done
-             done
+            # Get the subvolumegroup names from the filesystem name
+            ceph_command="ceph fs subvolumegroup ls ${fs}"
+            subvolgrp_name=$(timeout 120 oc -n "${ns}" exec "${HOSTNAME}"-helper -- bash -c "${ceph_command}"|awk -F\" '{print $4}')
```
Comment (Madhu-1) [inline must-gather/collection-scripts/gather_ceph_resources:338]: cant we use jq here instead of awk?

Comment (nik-redhat) [reply must-gather/collection-scripts/gather_ceph_resources]: Done

**Review by Madhu-1 — APPROVED** [review]
changes LGTM. please squash commits into 1

**Review by yati1998 — APPROVED** [review]

**Comment by yati1998** [comment]
> I doubt these tests are failing due to the changes in the PR thinking , is this a known issue?

Yeah it's fine

**Comment by nik-redhat** [comment]
If everything looks good, are we good to merge this PR?

**Comment by yati1998** [comment]
@agarwal-mudit

**Comment by nik-redhat** [comment]
@agarwal-mudit is the PR good to merge?

**Review by agarwal-mudit — APPROVED** [review]
/lgtm

**Pull request merged**